### PR TITLE
Clarify difference between Partial and Full verification

### PR DESCRIPTION
**Pull request by awwad**
### Purpose of the PR
- Clarifies the difference between Partial and Full verification in Primary, demo Primary, and demo Director code. This mostly revises docstrings, but also simplifies some demo code conditionals.
- In the demo, strips all CAN-related communication code intended for a CAN-based Partial verification Secondary.

These are in the same PR because of overlap, but they're fairly related. I also strip two unneeded global designations, a completely unrelated but miniscule change. :P

### Motivation
Initial motivation for this comes from the fact that this code seemed to be confusing to students. I'm much happier with the clarity of the result, but I'm sure there are related bits I can clarify. As for the CAN contents, less clutter is better, since that wasn't being used.

### CODE REVIEW

File: demo/demo_director.py [hunk 1]
```diff
@@ -517,8 +517,6 @@ def undo_sign_with_compromised_keys_attack():
     None.
   """
 
-  global director_service_instance
-
```
Comment (lukpueh) [inline demo/demo_director.py:521]: There are two more unneeded `global` statements in `demo_primary.py`, maybe elsewhere too?
But I guess fixing those is not really part of this PR. Worth a ticket? Also, I think its useful to mention in the commit message why you used globals in the first place and why they turned out to be unnecessary.

File: demo/demo_primary.py [hunk 2]
```diff
@@ -559,37 +559,27 @@ def get_image_for_ecu(ecu_serial):
...
-    uptane.Error
-      - if we are set to communicate via CAN interface, but CAN interface is
-        not ready
+  - For Full Verification Secondaries (the norm):
+      Send a zip archive of the most recent consistent set of the Primary's
+      client metadata directory, containing the current, consistent metadata
+      from all repositories used.
+
+  - For Partial Verification Secondaries:
+      Send the Director's Targets role file.
```
Comment (lukpueh) [inline demo/demo_primary.py:473]: Does it make sense to have an `<Arguments>` section and/or to mention that the argument `force_partial_verification` decides whether its a full or partial verification?

File: demo/demo_primary.py [hunk 3]
```diff
@@ -559,37 +559,27 @@ def get_image_for_ecu(ecu_serial):
 
 def get_metadata_for_ecu(ecu_serial, force_partial_verification=False):
   """
-  Send a zip archive of the most recent consistent set of the Primary's client
-  metadata directory, containing the current, consistent metadata from all
-  repositories used.
+  Provides the (updated) metadata a Secondary will need to validate updates.
```
Comment (lukpueh) [inline demo/demo_primary.py:463]: Is it worth mentioning what *updated* means?

**Comment by awwad** [comment]
The tree is getting a bit messy, so I'm going to merge this because the demos and tests all worked, and it's mostly documentation changes. I'll still want the review sometime, but there's no rush. (:

File: demo/demo_primary.py [hunk 4]
```diff
@@ -478,17 +421,14 @@ def enforce_jail(fname, expected_containing_dir):
...
-
-  For FV Secondaries via XMLRPC:
-    Returns:
-     - filename of the image, relative to the targets directory
-     - binary image data in xmlrpc.Binary format
+  Intended to be called via XMLRPC by the Secondary client, either partial or
+  full verification.
+
+  Returns the following to the requesting Secondary:
+     - filename of the firmware image assigned it by the Director and validated
```
Comment (lukpueh) [inline demo/demo_primary.py:428]: -it?

**Review by lukpueh — COMMENTED** [review]
Looks good to me.

File: demo/demo_primary.py (hunk 4 above)
Comment (awwad) [reply demo/demo_primary.py:428]: "it" here is the requesting Secondary.

Returns to X the filename assigned it by Y

Adding "to that Secondary" just seems too wordy. I guess the worst case is that people will think it's a typo and ignore it and probably still get it.

File: demo/demo_primary.py (hunk 2 above)
Comment (awwad) [reply demo/demo_primary.py:473]: It does, yes. Thanks.

File: demo/demo_director.py (hunk 1 above)
Comment (awwad) [reply demo/demo_director.py:521]: Fixed locally in 0bdaaf6. Will push with some other things.

File: demo/demo_primary.py (hunk 2 above)
Comment (awwad) [reply demo/demo_primary.py:473]: Fixed locally in d8324df. Will push with other things.

File: demo/demo_primary.py (hunk 3 above)
Comment (awwad) [reply demo/demo_primary.py:463]: Hmm. Replacing with current in 78e8d2c.

Adding these all in #148.
